sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ChangeFeedProcessorBuilder.java

```diff
@@ -119,6 +153,19 @@ public ChangeFeedProcessorBuilder options(ChangeFeedProcessorOptions changeFeedP
         return this;
     }
 
+    /**
+     * Sets the Change Feed Mode to be used with this {@link ChangeFeedProcessor}.
+     * Unless specifically set, the default will be {@link ChangeFeedMode#LATEST_VERSION} to support backward-compatibility.
+     *
+     * @param changeFeedMode changeFeedMode to be used for this {@link ChangeFeedProcessor}
+     * @return current Builder
+     */
+    @Beta(value = Beta.SinceVersion.V4_37_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ChangeFeedProcessorBuilder changeFeedMode(ChangeFeedMode changeFeedMode) {
```

Comment:
do we still need the changeFeedMode in the public API? 
I thought  handleAllVersionsAndDeletesChanges -> ALL_VERSIONS_AND_DELETES
handleChanges -> LATEST_VERSION?

Reply:
We still need to expose ChangeFeedMode for pull model correct? For change feed processor i think that proposla makes sense - although I would just rather introduce an overload of handleChanges with an flag to opt-in into new wire protocol
But both work for me

Reply:
Discussed offline, we don't need to introduce `ChangeFeedMode` yet to public as we can detect the change feed mode from the new handle API that we are exposing.